lib/resource-providers/vpc.ts

```diff
@@ -29,9 +35,32 @@ export class VpcProvider implements ResourceProvider<ec2.IVpc> {
             // It will automatically divide the provided VPC CIDR range, and create public and private subnets per Availability Zone.
             // Network routing for the public subnets will be configured to allow outbound access directly via an Internet Gateway.
             // Network routing for the private subnets will be configured to allow outbound access via a set of resilient NAT Gateways (one per AZ).
+            // Creates Secondary CIDR and Secondary subnets if passed.
             vpc = new ec2.Vpc(context.scope, id + "-vpc");
+            const secondarySubnets: Array<PrivateSubnet> = [];
+            if (this.secondaryCidr) {
+                new ec2.CfnVPCCidrBlock(context.scope, id + "-secondaryCidr", {
+                    vpcId: vpc.vpcId,
+                    cidrBlock: this.secondaryCidr});
+                if (this.secondarySubnetCidrs) {
+                    for (let i = 0; i < vpc.availabilityZones.length; i++) {
+                        if (this.secondarySubnetCidrs[i]) {
+                            secondarySubnets[i] = new ec2.PrivateSubnet(context.scope, id + "private-subnet-" + i, {
```

Comment:
if it is fully private subnet, will pods deployed here not have access to the internet since I don't see outbound route table defined here? Won't it mean that external docker images will fail?

Reply:
https://docs.aws.amazon.com/eks/latest/userguide/cni-custom-network.html#:~:text=The%20nodes%20are%20configured%20in%20public%20subnets%2C%20and%20you%20want%20to%20place%20the%20pods%20in%20private%20subnets.%20The%20route%20table%20associated%20to%20a%20public%20subnet%20includes%20a%20route%20to%20an%20internet%20gateway.%20The%20route%20table%20associated%20to%20a%20private%20subnet%20doesn%27t%20include%20a%20route%20to%20an%20internet%20gateway. 

Nodes Interface will be used to access the internet and thats primarily the usecase for using custom networking with pods in Private subnet with not internet access and nodes can be in public or private subnet with access to NAT.